[feat/fix]: Improve the contributing guidelines (#1503)
TheAlgorithms/C-Plus-Plus: CONTRIBUTING.md

### Contributor

> We are happy that you consider implementing algorithms and data structures for others! This repository is referred to and used by learners from around the globe. Being one of our contributors, you agree and confirm that:

Reviewer: ```suggestion
Being a contributor at The Algorithms we request you to follow the points mentioned below.
```

- You did your own work.

>   - No plagiarism allowed. Any plagiarized work will not be merged.

Reviewer: - Please try to provide your own implementations with proper documentation. We do not entertain any plagiarized work.

Author: I don't think this is needed. The current paragraph is OK IMO.

> - Your work will be distributed under [MIT License](License) once your pull request has been merged.

Reviewer: This looks good.

> - Your submitted work fulfills or mostly fulfills our styles and standards.

Reviewer: ```suggestion
- Please follow the repository coding guidelines (mentioned below)
```

Reviewer: Also add one more point
- Please make sure that you do not provide any duplicate work (in a sense that implementation of data structure or algorithm is already present in repository)

Author: > Suggestion

What about `Please follow the repository guidelines and standards mentioned below`?

**New implementation** New implementations are welcome!

> **Improving comments** and **adding tests** to existing algorithms are much appreciated.

Reviewer: - You can add new algorithms or data structures which are not present in repository or can improve the old implementations (documentation, improving test cases, removing bugs or in any other resonable sense)

> **Issues** Please avoid opening issues asking to be "assigned” to a particular algorithm. This merely creates unnecessary noise for maintainers. Instead, please submit your implementation in a pull request, and it will be evaluated by project maintainers.

Reviewer: This looks good.

### Making Changes

#### Code

- Please use the directory structure of the repository.

> - File extension for the code should be `*.hpp`, `*.h` or `*.cpp`.

Reviewer: ```suggestion
- Make sure file extensions should be `*.hpp`, `*.h` or `*.cpp`.
```

- Don't use **`bits/stdc++.h`** because this is quite Linux-specific and slows down the compilation process.
- Organize your code using **`struct`**, **`class`**, and/or **`namespace`** keywords
- If an implementation of the algorithm already exists, please refer to the [file-name section below](#new-file-name-guidelines).
- You can suggest reasonable changes to existing algorithms.
- Strictly use snake_case (underscore_separated) in filenames.
- If you have added or modified code, please make sure the code compiles before submitting.
- Our automated testing runs [__CMake__](https://cmake.org/) on all the pull requests, so please be sure that your code passes before submitting.
- Please conform to [Doxygen](https://www.doxygen.nl/manual/docblocks.html) standard and document the code as much as possible. This not only facilitates the readers but also generates the correct info on the website.

> - **Be consistent in the use of these guidelines.**

Reviewer: Looks good

#### Documentation

- Make sure you put useful comments in your code. Do not comment on obvious things.
- Please avoid creating new directories if at all possible. Try to fit your work into the existing directory structure. If you want to create a new directory, then please check if a similar category has been recently suggested or created by other pull requests.

> - If you have modified/added documentation, please ensure that your language is concise and contains no grammar errors.

Reviewer: ```suggestion
- If you have modified/added documentation, please ensure that your language is concise and must not contain gramatical errors.
```
